cmd/kubeadm/app/constants/constants.go

```diff
@@ -121,6 +121,9 @@ var (
 
 	// MinimumControlPlaneVersion specifies the minimum control plane version kubeadm can deploy
 	MinimumControlPlaneVersion = version.MustParseSemantic("v1.7.0")
+
+	// MinimumCSRAutoApprovalClusterRolesVersion
+	MinimumCSRAutoApprovalClusterRolesVersion = version.MustParseSemantic("v1.8.0-alpha.3")
```

Comment:
I really wish there was a better way to handle new features as we shouldn't be gating on it until some reasonable level of testing.  Could you add a //TODO to convert to 1.8.0 once we near release.